security/security.c

```diff
@@ -436,28 +436,25 @@ static void __init lsm_init_hook_static_slot(struct static_slot *slots,
 					     int *first_slot_idx)
 {
 	struct security_hook_list *pos;
-	struct static_slot *slot;
-	int slot_cnt;
+	int slot_cnt, slot_idx;
 
 	slot_cnt = 0;
 	hlist_for_each_entry_rcu(pos, head, list)
 		slot_cnt++;
 
-	if (slot_cnt > SECURITY_STATIC_SLOT_COUNT)
-		panic("%s - No static hook slot remaining to add LSM hook.\n",
-		      __func__);
-
 	if (slot_cnt == 0) {
 		*first_slot_idx = INT_MAX;
 		return;
 	}
 
 	*first_slot_idx = SECURITY_STATIC_SLOT_COUNT - slot_cnt;
-	slot = slots + *first_slot_idx;
+	slot_idx = *first_slot_idx;
 	hlist_for_each_entry_rcu(pos, head, list) {
-		__static_call_update(slot->key, slot->trampoline,
-				     pos->hook.generic_func);
-		slot++;
+		if (slot_idx >= 0)
```

Comment:
Shouldn't this be `if slot_idx < SECURITY_STATIC_SLOT_COUNT`?

Reply:
I didn't provide enough context here: the slots will hold the callbacks that are at the end of the linked list. This way, we can iterate over the first few elements of the list which are not in the slots.
E.g. if we have 10 slots and 15 callbacks, `slot_idx` will be -5 at first. It will skip the first 5 elements in the list, and then start adding the callbacks to the slots.